Gems/ROS2/Code/Source/Manipulation/FollowJointTrajectoryActionServer.cpp

```diff
@@ -132,6 +108,42 @@ namespace ROS2
     void FollowJointTrajectoryActionServer::GoalAcceptedCallback(const std::shared_ptr<GoalHandle> goalHandle)
     {
         AZ_Trace("FollowJointTrajectoryActionServer", "Goal accepted\n");
+
+        if (!IsReadyForExecution())
+        {
+            AZ_Trace("FollowJointTrajectoryActionServer", "Goal aborted: server is not ready for execution!");
+            if (m_goalHandle)
+            {
+                AZ_Trace(
+                    "FollowJointTrajectoryActionServer",
+                    " is_active: %d,  is_executing %d, is_canceling : %d",
+                    m_goalHandle->is_active(),
+                    m_goalHandle->is_executing(),
+                    m_goalHandle->is_canceling());
+            }
+
+            auto result = std::make_shared<FollowJointTrajectory::Result>();
+            result->error_string = "Goal aborted: server is not ready for execution!";
+            result->error_code = FollowJointTrajectory::Result::INVALID_GOAL;
+            goalHandle->abort(result);
+            return;
+        }
+
+        AZ::Outcome<void, AZStd::string> executionOrderOutcome;
+        JointsTrajectoryRequestBus::EventResult(
+            executionOrderOutcome, m_entityId, &JointsTrajectoryRequests::StartTrajectoryGoal, goalHandle->get_goal());
+
+        if (!executionOrderOutcome)
+        {
+            AZ_Trace("FollowJointTrajectoryActionServer", "Execution not be accepted: %s", executionOrderOutcome.GetError().c_str());
+
+            auto result = std::make_shared<FollowJointTrajectory::Result>();
+            result->error_string = "Execution not accepted: " + std::string(executionOrderOutcome.GetError().c_str());
+            result->error_code = FollowJointTrajectory::Result::INVALID_GOAL;
```

Comment:
It is better to propagate result from the outcome -> here, the correct result would actually be INVALID_JOINTS (since the call earlier only validates that), but it should not depend on magic knowledge of what is happening.

Reply:
I changed StartTrajectoryGoal and TrajectoryGoal to return FollowJointTrajectory::Response in case of failure, as it includes error_code, not only string.